### PR TITLE
Download and install (locally) ninja.

### PR DESCRIPTION
**Pull request by guruDanny67**
This seems easy ...

The unzip program is been added to the msys2 requirement.

I call exec_msys(...) while maybe it's been better to call directly __execute(...) but it's local to the builder class. 

We can avoid the check of the presence of the unzip.exe and call exec_msys('unzip -o ...' ) but I think in this way is clearer what happens and maybe the unzip is needed in teh future.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -203,6 +203,31 @@ def get_dict():
...
+        # We download a .zip file so we estract it in the tool directory ...
+        destdir = os.path.join(self.builder.opts.tools_root_dir, 'ninja')
+        destfile = os.path.join(destdir, 'ninja.exe')
+        if not os.path.isfile(destfile):
+            print_log("Unpacking ninja le to tools directory (%s)" % (destfile, ))
+            self.builder.make_dir(destdir)
+            self.builder.exec_msys('%s -o %s -d %s' % (self.builder.unzip, self.archive_file, destdir, ))
+        # .. and set the builder object to point to the file
+        self.builder.ninja = destfile
+        pass
```
Comment (nacho) [inline build.py]: I'd say remove this pass, it is not needed

**Comment by guruDanny67** [comment]
Done, my mistake from the cut & paste of nuget

**Comment by guruDanny67** [comment]
I'll try. If things goes wrong i follow the xkcd 1597 advices :)

**Comment by guruDanny67** [comment]
Actually I'm been able to make a single commit but on another branch ....

Step by step I'll learn to use git better.

I put another pull request and I'm closing this one, sorry.

**Pull request closed without merging**